### PR TITLE
Configure --with-modules when EMACS_VERSION >= 25

### LINKED ISSUE #4: Enable modules support for Emacs > 25

### PR DESCRIPTION
**Pull request by vermiculus**
Fix #4

I'm not sure how to test these changes without actaully using it on Travis -- is there a better way?

### CODE REVIEW

**Comment by vermiculus** [comment]
Not sure what's going on with the PRETEST build; can someone take a look? Looks like 26.0.91 *does* exist on alpha.

**Comment by vermiculus** [comment]
Actually, it might just need to be restarted. #15 is working swimmingly, but it includes this commit.

**Comment by vermiculus** [comment]
Ah, yeah, `install_gnutls failed`: https://travis-ci.org/flycheck/emacs-travis/jobs/388306115#L837. Just needs to be restarted 😄

File: emacs-travis.mk [hunk 1]
```diff
@@ -42,6 +39,17 @@ MAJOR_VERSION := $(word 1,$(subst ., ,$(EMACS_VERSION)))
 # Whether the version is a release candidate
 PRETEST ?= $(findstring rc,$(PRE_RELEASE_PART))
 
+# Build a minimal Emacs with no special flags, to build as fast as possible
+ifndef EMACSCONFFLAGS
+EMACSCONFFLAGS := --with-x-toolkit=no --without-x --without-all --with-xml2 \
+	CFLAGS='-O2 -march=native' \
+	CXXFLAGS='-O2 -march=native'
+
+ifeq ($(shell test ( $(EMACS_VERSION) -eq snapshot ) -o ( $(MAJOR_VERSION) -ge 25 ); echo $$?),0)
```
Comment (fmdkdd) [inline emacs-travis.mk]: So you are calling `test` and checking the status code against 0, is that it?

I suppose there's no better way?

File: emacs-travis.mk [hunk 2]
```diff
@@ -42,6 +39,17 @@ MAJOR_VERSION := $(word 1,$(subst ., ,$(EMACS_VERSION)))
 # Whether the version is a release candidate
 PRETEST ?= $(findstring rc,$(PRE_RELEASE_PART))
 
+# Build a minimal Emacs with no special flags, to build as fast as possible
+ifndef EMACSCONFFLAGS
```
Comment (fmdkdd) [inline emacs-travis.mk]: Why the `ifndef` here?  Is it because `+=` is not compatible with `?=`?

Comment (vermiculus) [reply emacs-travis.mk]: Yep! `?=` is sugar for `ifndef`, so using the latter gives us some more flexibility.

File: emacs-travis.mk (hunk 1 above)
Comment (vermiculus) [reply emacs-travis.mk]: You could take the approach taken in d00f0b7481c8e929be218d885a2281415f44bfc8 to update GnuTLS on 26.1 and snapshot, but I prefer this approach. While it's a little more verbose, there's no duplication of logic.

**Comment by vermiculus** [comment]
magit/libegit2#8 is building a libgit2 module for magit; testing involves loading the module and doing some sanity checks.

**Comment by fmdkdd** [comment]
> magit/libegit2#8 is building a libgit2 module for magit; testing involves loading the module and doing some sanity checks.

Oh great!

Thank you for the answers.

**Comment by vermiculus** [comment]
Thanks!!

**Pull request merged**